pkg/update/check.go

```diff
@@ -0,0 +1,119 @@
+package update
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"runtime"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/aquasecurity/trivy/pkg/log"
+)
+
+type versionInfo struct {
+	LatestVersion string    `json:"latest_version"`
+	LatestDate    time.Time `json:"latest_date"`
+}
+
+type announcement struct {
+	FromDate     time.Time `json:"from_date"`
+	ToDate       time.Time `json:"to_date"`
+	Announcement string    `json:"announcement"`
+}
+
+type updateResponse struct {
+	Trivy         versionInfo    `json:"trivy"`
+	Announcements []announcement `json:"announcements"`
+	Warnings      []string       `json:"warnings"`
+}
+
+var (
+	updatesApi       = "https://api.trivy.cloud/check"
+	responseRecieved atomic.Bool
+	currentVersion   string
+	latestVersion    updateResponse
+)
+
+// CheckUpdate makes a best efforts request to determine the
+// latest version and any announcements
+func CheckUpdate(ctx context.Context, version string, args []string) {
+	currentVersion = version
+
+	go func() {
+		args = getFlags(args)
+
+		log.Debug("[version] Requesting latest details")
```

Comment:
You can use `WithPrefix`:
```suggestion
	logger := log.WithPrefix("version")
	go func() {
		args = getFlags(args)

		logger.Debug("Requesting latest details")
```

Reply:
ah good to know, thanks

Reply:
I also suggest to use "notice" prefix. Version is not good here i think.

Reply:
set to `version-check` for now